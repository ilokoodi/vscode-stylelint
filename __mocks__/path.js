'use strict';

const path = jest.requireActual('path');

/**
 * Mock platform.
 * @type {'posix' | 'win32'}
 */
let mockPlatform = 'posix';

/**
 * Sets the mock platform.
 * @param {'posix' | 'win32'} platform
 * @returns {void}
 */
const __mockPlatform = (platform) => {
	mockPlatform = platform;
};

const pathProxy = new Proxy(path, {
	get(_, name) {
		if (name === '__mockPlatform') {
			return __mockPlatform;
		}

		if (mockPlatform === 'win32') {
			if (typeof path.win32[name] === 'function') {
				return jest.fn(path.win32[name]);
			}

			return path.win32[name];
		}

		if (typeof path.posix[name] === 'function') {
			return jest.fn(path.posix[name]);
		}

		return path.posix[name];
	},
});

module.exports = pathProxy;
