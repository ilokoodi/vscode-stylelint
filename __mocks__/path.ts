const path = jest.requireActual('path');

/**
 * Mock platform.
 */
let mockPlatform: 'posix' | 'win32' | undefined = undefined;

/**
 * Sets the mock platform. A value of `undefined` will use the actual platform.
 */
const __mockPlatform = (platform?: 'posix' | 'win32'): void => {
	mockPlatform = platform;
};

const mockedFns: {
	default: Record<string | symbol, () => unknown>;
	posix: Record<string | symbol, () => unknown>;
	win32: Record<string | symbol, () => unknown>;
} = {
	default: {},
	posix: {},
	win32: {},
};

const pathProxy = new Proxy(path, {
	get(_, name) {
		if (name === '__mockPlatform') {
			return __mockPlatform;
		}

		if (!mockPlatform) {
			if (typeof path[name] === 'function') {
				if (mockedFns.default[name]) {
					return mockedFns.default[name];
				}

				mockedFns.default[name] = jest.fn(path[name]);

				return mockedFns.default[name];
			}

			return path[name];
		}

		if (mockPlatform === 'win32') {
			if (typeof path.win32[name] === 'function') {
				if (mockedFns.win32[name]) {
					return mockedFns.win32[name];
				}

				mockedFns.win32[name] = jest.fn(path.win32[name]);

				return mockedFns.win32[name];
			}

			return path.win32[name];
		}

		if (typeof path.posix[name] === 'function') {
			if (mockedFns.posix[name]) {
				return mockedFns.posix[name];
			}

			mockedFns.posix[name] = jest.fn(path.posix[name]);

			return mockedFns.posix[name];
		}

		return path.posix[name];
	},
});

export = pathProxy;
