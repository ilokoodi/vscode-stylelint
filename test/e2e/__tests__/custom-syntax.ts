import path from 'path';
import pWaitFor from 'p-wait-for';
import { workspace, commands, window } from 'vscode';

describe('vscode-stylelint with "stylelint.customSyntax"', () => {
	it('auto-fix should work if "stylelint.customSyntax" is enabled', async () => {
		const cssDocument = await workspace.openTextDocument(
			path.resolve(workspaceDir, 'custom-syntax/test.css'),
		);

		await window.showTextDocument(cssDocument);

		await pWaitFor(
			async () => {
				const names = await commands.getCommands();

				return (
					// cspell:disable-next-line
					names.includes('stylelint.executeAutofix') && names.includes('stylelint.applyAutoFix')
				);
			},
			{ timeout: 2000 },
		);

		// cspell:disable-next-line
		await commands.executeCommand('stylelint.executeAutofix');

		expect(cssDocument.getText()).toMatchSnapshot();
	});
});
