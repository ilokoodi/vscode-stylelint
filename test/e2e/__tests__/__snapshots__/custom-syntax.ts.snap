// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vscode-stylelint with "stylelint.customSyntax" auto-fix should work if "stylelint.customSyntax" is enabled 1`] = `
"/* prettier-ignore */
.foo .bar
    color: red"
`;
